src/spetlr/configurator/configurator.py

```diff
@@ -302,12 +320,24 @@ def get_unique_id_length(self):
         """
         return len(self._unique_id)
 
-    def register(self, key: str, value: TcValue):
+    def register(self, key: str, value: TcValue) -> str:
         """
-        Register a new item.
+        Register a new item and return its key.
         """
         self._raw_resource_details[key] = value
         self.table_details = dict()
+        return key
+
+    def define(self, **kwargs) -> str:
```

Comment:
Do we need to add further checks of the input? Can there be an case where the kwargs does not have the correct format? (I ask since I am not experienced in using the kwargs input type)